resources/google_agent.rb

```diff
@@ -0,0 +1,177 @@
+# Copyright:: 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# To learn more about Custom Resources, see https://docs.chef.io/custom_resources/
+# Use upgrade because it'll fall back to install if software isn't installed already
+# By using upgrade as the default action, if the admin changes the version attribute,
+# then the software will update. If this was set to just install, changing the version
+# attribute would not result in servers being upgraded.
+default_action :upgrade
+resource_name :google_agent
+provides :google_agent
+unified_mode true
+
+# Install software, at the specified version
+action :install do
+  case node['platform_family']
+  when 'rhel', 'debian', 'suse', 'amazon'
+    bash node['file_url_name'] do
+      cwd node['tmp_file_path']
+      code "./#{node['file_url_name']} --also-install --version=#{node['version']}"
+      only_if "./#{node['file_url_name']} --also-install --version=#{node['version']} --dry-run | grep -qi 'installation succeeded'" == 0
+      user 'root'
+    end
+    case node['final_agent_type']
+    when 'google-cloud-ops'
```

Comment:
Similar question here: what's the reason behind having both `google-cloud-ops` and `ops-agent`?